cluster-provision/okd/scripts/provision.sh

```diff
@@ -90,31 +90,136 @@ if [ ! -f "/etc/installer/token" ]; then
     exit 1
 fi
 
-mkdir -p /root/install
-cp /install-config.yaml /root/install/
+export CLUSTER_DIR=/root/install
+INSTALL_CONFIG_FILE=$CLUSTER_DIR/install-config.yaml
+
+# we need to install jq and yq to make yaml changes
+dnf install -y jq
+pip install yq
+
+mkdir -p $CLUSTER_DIR
+
+# print version
+/openshift-install version
+
+if [[ $INSTALLER_TAG =~ ^.*4\.1$ ]]; then
+    cp /install-config.yaml $CLUSTER_DIR/
+
+elif [[ $INSTALLER_TAG =~ ^.*4\.2$ ]]; then
+
+    # modify number of workers
+    yq -y '.compute[].replicas = 2' /install-config.yaml > "$INSTALL_CONFIG_FILE"
+
+else
+    echo "No case defined for $INSTALLER_TAG" >&2
+    exit 1
+fi
 
 # inject pull secret into install config
-cat /etc/installer/token >> /root/install/install-config.yaml
+cat /etc/installer/token >> $CLUSTER_DIR/install-config.yaml
 
 # inject vagrant ssh public key into install config
 ssh_pub_key="ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key"
-echo "sshKey: '$ssh_pub_key'" >> /root/install/install-config.yaml
+echo "sshKey: '$ssh_pub_key'" >> $CLUSTER_DIR/install-config.yaml
+
+# Generate manifests
+/openshift-install create manifests --dir=$CLUSTER_DIR
+
+# increase workers memory to 6144MB
+sed -i -e "s/domainMemory: 4096/domainMemory: $WORKERS_MEMORY/" $CLUSTER_DIR/openshift/99_openshift-cluster-api_worker-machineset-0.yaml
+sed -i -e "s/domainVcpu: 2/domainVcpu: $WORKERS_CPU/" $CLUSTER_DIR/openshift/99_openshift-cluster-api_worker-machineset-0.yaml
+
+if [[ $INSTALLER_TAG =~ ^.*4\.2$ ]]; then
+
+    # generate machineconfig for insecure-registries beforehand
+
+    cat > registries.conf << __EOF__
+[registries]
+  [registries.search]
+    registries = ["registry.access.redhat.com", "docker.io"]
+  [registries.insecure]
+    registries = ["brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888", "registry:5000"]
```

Comment:
Do we still need the brew pulp registry, isn't those containers at quay now ? At least at CNV they have move it from pulp to quay.